library/hls/src/main/java/com/google/android/exoplayer2/source/hls/HlsChunkSource.java

```diff
@@ -270,28 +270,35 @@ public long getAdjustedSeekPositionUs(long positionUs, SeekParameters seekParame
       return positionUs;
     }
 
+    long startOfPlaylistInPeriodUs =
+        mediaPlaylist.startTimeUs - playlistTracker.getInitialStartTimeUs();
+    long targetPositionInPlaylistUs = positionUs - startOfPlaylistInPeriodUs;
+
+    if (targetPositionInPlaylistUs > mediaPlaylist.durationUs) {
+      return positionUs;
+    }
+
     // Segments start with sync samples (i.e., EXT-X-INDEPENDENT-SEGMENTS is set) and the playlist
-    // is non-empty, so we can use segment start times as sync points. Note that in the rare case
-    // that (a) an adaptive quality switch occurs between the adjustment and the seek being
+    // is non-empty, and the playlist is fresh enough that the period position falls in side of
+    // the playlist bound, so we can use segment start times as sync points.
+    //
+    // If (a) an adaptive quality switch occurs between the adjustment and the seek being
     // performed, and (b) segment start times are not aligned across variants, it's possible that
     // the adjusted position may not be at a sync point when it was intended to be. However, this is
     // very much an edge case, and getting it wrong is worth it for getting the vast majority of
     // cases right whilst keeping the implementation relatively simple.
-    long startOfPlaylistInPeriodUs =
-        mediaPlaylist.startTimeUs - playlistTracker.getInitialStartTimeUs();
-    long relativePositionUs = positionUs - startOfPlaylistInPeriodUs;
-    int segmentIndex =
+    int segIndex =
         Util.binarySearchFloor(
             mediaPlaylist.segments,
-            relativePositionUs,
+            targetPositionInPlaylistUs,
```

Comment:
Mixing renaming, code changes and reformatting really makes it confusing for us to maintain.   Please let us know any formatting or other restructuring changes before the pull request is approved so the "merge" commit can be pure and free from other changes.

Reply:
P.S. This  change to move `startOfPlaylistInPeriodUs` add to convert playlist time to period time at the end is much more clear, thanks.

I'm happy to debate the name, but `targetPositionInPlaylistUs` seems more descriptive to me then `relativePositionUs`